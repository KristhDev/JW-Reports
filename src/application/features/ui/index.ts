export { default as uiReducer, UI_INITIAL_STATE, setIsDataExporting, setIsKeyboardVisible, setOldDatetimePicker } from './slice';
export * from './types';