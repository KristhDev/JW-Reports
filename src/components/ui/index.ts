export { Button } from './Button';
export { EyeBtn } from './EyeBtn';
export { FormField } from './FormField';
export { Title } from './Title';