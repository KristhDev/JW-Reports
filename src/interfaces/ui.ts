export type NavigationParamsList = {
    MainTabsBottomNavigation: undefined;
    SettingsStackNavigation: undefined;
    ProfileScreen: undefined;
    ResetPasswordScreen: undefined;
    AuthStackNavigation: undefined;
}

export type MainTabsBottomParamsList = {
    PreachingStackNavigation: undefined;
    RevistsStackNavigation: undefined;
    CoursesStackNavigation: undefined;
}

export interface LoadResourcesOptions {
    loadMore?: boolean;
    refresh?: boolean;
    search?: string;
}

export interface Pagination {
    from: number;
    to: number;
}

export interface ModalProps {
    isOpen: boolean;
    onClose: () => void;
}

export type ItemOption = {
    label: string;
    value: string;
}