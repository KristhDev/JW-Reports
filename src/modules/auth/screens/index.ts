export { Credentials } from './Credentials';
export { ForgotPassword } from './ForgotPassword';
export { Login } from './Login';
export { Profile } from './Profile';
export { Register } from './Register';