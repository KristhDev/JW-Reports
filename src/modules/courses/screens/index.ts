export { ActiveOrSuspendCourseModal } from './ActiveOrSuspendCourseModal';
export { AddOrEditCourse } from './AddOrEditCourse';
export { CourseDetail } from './CourseDetail';
export { Courses } from './Courses';
export { FinishOrStartCourseModal } from './FinishOrStartCourseModal';
export { PassToCourseModal } from './PassToCourseModal';