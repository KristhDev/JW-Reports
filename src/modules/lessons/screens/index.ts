export { AddOrEditLesson } from './AddOrEditLesson';
export { FinishOrStartLessonModal } from './FinishOrStartLessonModal';
export { LessonDetail } from './LessonDetail';
export { Lessons } from './Lessons';