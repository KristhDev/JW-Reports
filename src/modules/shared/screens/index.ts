export * from './settings';
export * from './status';