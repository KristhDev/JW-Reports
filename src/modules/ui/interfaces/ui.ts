export interface UIState {
    isKeyboardVisible: boolean;
}

export type SetIsVisiblePayload = {
    isVisible: boolean;
}

/**
 * NavigationParamsList is a type that is an object with a bunch of properties, each of which is a type
 * that is an object with a single property called undefined.
 *
 * @property MainTabsBottomNavigation - undefined;
 * @property SettingsStackNavigation - undefined;
 * @property ProfileScreen - undefined;
 * @property ResetPasswordScreen - undefined;
 * @property AuthStackNavigation - undefined;
 */
export type NavigationParamsList = {
    MainTabsBottomNavigation: undefined;
    SettingsStackNavigation: undefined;
    ProfileScreen: undefined;
    ResetPasswordScreen: undefined;
    AuthStackNavigation: undefined;
}

/**
 * MainTabsBottomParamsList is a type that is an object with three properties, each of which is a type
 * that is a function that takes undefined and returns undefined.
 *
 * @property PreachingStackNavigation - undefined;
 * @property RevisitsStackNavigation - undefined;
 * @property CoursesStackNavigation - undefined;
 */
export type MainTabsBottomParamsList = {
    PreachingStackNavigation: undefined;
    RevisitsStackNavigation: undefined;
    CoursesStackNavigation: undefined;
}

/**
 * Defining an interface called LoadResourcesOptions.
 *
 * @property {boolean} loadMore - If the load more button should be shown.
 * @property {boolean} refresh - If the refresh button should be shown.
 * @property {string} search - The search term.
 */
export interface LoadResourcesOptions {
    loadMore?: boolean;
    refresh?: boolean;
    search?: string;
}

/**
 * Defining an interface called Pagination.
 *
 * @property {number} from - The number of items to skip.
 * @property {number} to - The number of items to return.
 */
export interface Pagination {
    from: number;
    to: number;
}

/**
 * Defining an interface called ModalProps.
 *
 * @property {boolean} isOpen - If the modal is open or not.
 * @property {() => void} onClose - The function that will be called when the modal is closed.
 */
export interface ModalProps {
    isOpen: boolean;
    onClose: () => void;
}

/**
 * ItemOption is an object with a label property of type string and a value property of type string.
 *
 * @property {string} label - The text that will be displayed in the dropdown
 * @property {string} value - The value of the item.
 */
export type ItemOption = {
    label: string;
    value: string;
}

/**
 * Defining an interface called StorageError.
 *
 * @property {string} message - The error message.
 * @property {string} name - The error name.
 * @property {string} stack - The error stack.
 */
export interface StorageError {
    message: string;
    name: string;
    stack?: string;
}
