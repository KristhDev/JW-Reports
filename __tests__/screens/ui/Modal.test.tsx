import React from 'react';
import { View } from 'react-native';
import { render, screen } from '@testing-library/react-native';

import { Modal } from '../../../src/screens/ui';

describe('Test in <Modal /> screen', () => {
    beforeEach(() => {
        render(
            <Modal isOpen>
                <View
                    style={{
                        backgroundColor: '#FFF',
                        height: 200,
                        width: 300
                    }}
                    testID="modal-content"
                />
            </Modal>
        );
    });

    it('should to match snapshot', () => {
        expect(screen.toJSON()).toMatchSnapshot();
    });

    it('should render children', () => {
        const children = screen.getByTestId('modal-content');
        expect(children).toBeTruthy();
    });
});