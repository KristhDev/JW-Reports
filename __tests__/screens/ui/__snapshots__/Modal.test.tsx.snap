// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <Modal /> screen should to match snapshot 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  statusBarTranslucent={true}
  transparent={true}
  visible={true}
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.5)",
        "flex": 1,
        "justifyContent": "center",
        "zIndex": 999,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#FFF",
          "height": 200,
          "width": 300,
        }
      }
      testID="modal-content"
    />
  </View>
</Modal>
`;
